components/eam/src/chemistry/modal_aero/modal_aero_amicphys_subareas.F90

```diff
@@ -0,0 +1,649 @@
+                                                           qsub3(:,jclea), qsub3(:,jcldy)    )! inout
+
+     !=========================================================================
+     ! Set AEROSOL mixing ratios in subareas.
+     ! Only need to do this for points 2 and 3 in the time integraion loop,
+     ! i.e., the before-cloud-chem and after-cloud-chem states.
+     !=========================================================================
+     ! Cloud-borne aerosols. (They are straightforward to partition, 
+     ! as they only exist in the cloudy subarea.)
+     !----------------------------------------------------------------------------------------
+     ! Partition mass and number before cloud chemistry
+
+     call set_subarea_qnumb_for_cldbrn_aerosols( loffset, jclea, jcldy, fcldy, qqcwgcm2, qqcwsub2 )
+     call set_subarea_qmass_for_cldbrn_aerosols( loffset, jclea, jcldy, fcldy, qqcwgcm2, qqcwsub2 )
+
+     ! Partition mass and number before cloud chemistry
+
+     call set_subarea_qnumb_for_cldbrn_aerosols( loffset, jclea, jcldy, fcldy, qqcwgcm3, qqcwsub3 )
+     call set_subarea_qmass_for_cldbrn_aerosols( loffset, jclea, jcldy, fcldy, qqcwgcm3, qqcwsub3 )
+
+     !----------------------------------------------------------------------------------------
+     ! Interstitial aerosols. (They can exist in both cloudy and clear subareas, and hence 
+     ! need to be partitioned.)
+     !----------------------------------------------------------------------------------------
+     ! Partition mass and number before cloud chemistry
+
+     call set_subarea_qnumb_for_intrst_aerosols( loffset, jclea, jcldy, fclea, fcldy, &! in
+                                                 qgcm2, qqcwgcm2,                     &! in
+                                                 qgcm2, qsub2                         )! in, inout
+
+     call set_subarea_qmass_for_intrst_aerosols( loffset, jclea, jcldy, fclea, fcldy, &! in
+                                                 qgcm2, qqcwgcm2,                     &! in
+                                                 qgcm2, qsub2                         )! in, inout
+
+     ! Partition mass and number before cloud chemistry
+
+     call set_subarea_qnumb_for_intrst_aerosols( loffset, jclea, jcldy, fclea, fcldy, &! in
+                                                 qgcm2, qqcwgcm2,                     &! in
+                                                 qgcm3, qsub3                         )! in, inout
+
+     call set_subarea_qmass_for_intrst_aerosols( loffset, jclea, jcldy, fclea, fcldy, &! in
+                                                 qgcm2, qqcwgcm2,                     &! in
+                                                 qgcm3, qsub3                         )! in, inout
+
+  end if ! different categories
+
+end subroutine set_subarea_gases_and_aerosols
+
+subroutine copy_cnst( q_in, q_copy, lcopy )
```

Comment:
@huiwanpnnl , do you think this subroutine is more clear than the `where` clause that does the copy itself? Are people confused by the use of `where`? To me, this is a simple construction that I wish other programming languages had.

Reply:
Ah, I see what you mean.

This subroutine `copy_cnst` is a remnant of several rounds of revision. Earlier, it had a longer name and several additional input arguments. Now that it has become this simple, we can simple write `where (lcopy) q_copy = q_in` in the calling routine. I removed the `copy_cnst` subroutine in [1b769fd](https://github.com/eagles-project/e3sm_mam4_refactor/pull/49/commits/1b769fdf84fcb3e4f395d22cd06225b4b54b1f98).

Reply:
Thanks for the historical perspective, Hui!